modules/Rock.cmake

```diff
@@ -49,6 +49,30 @@ function(rock_add_compiler_flag_if_it_exists FLAG)
     endif()
 endfunction()
 
+macro(rock_add_target_properties _target _name _properties)
+    set(_new_properties ${_properties} ${ARGN})
+    get_target_property(_old_properties ${_target} ${_name})
+    if(NOT _old_properties)
+        set(_old_properties)
+    else()
+        set(_old_properties "${_old_properties} ")
+    endif()
+    string (REPLACE ";" " " _new_properties "${_old_properties}${_new_properties}")
```

Comment:
This won't work because you cannot pass a list to set_target_properties. You have to explicitly convert it to string and as you said, cmake will separate elements with semicolons which will be passed to the compiler and everything will obviously break. We either drop the ${ARGN} approach and force the user to pass a single argument to our macro or explicitly remove the semicolons from the converted string.